sor/src/test/java/com/bazaarvoice/emodb/sor/audit/s3/AthenaAuditWriterTest.java

```diff
@@ -0,0 +1,184 @@
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Map<String, Object> auditJson = JsonHelper.fromJson(line, new TypeReference<Map<String, Object>>() {});
+                    _uploadedAudits.put((String) invocationOnMock.getArguments()[1], auditJson);
+                }
+            }
+
+            PutObjectResult result = new PutObjectResult();
+            result.setETag(file.getName());
+            return result;
+        });
+
+        _tempStagingDir = Files.createTempDir();
+
+        // Start with some default time; individual tests can override as necessary
+        _now = Instant.from(ZonedDateTime.of(2018, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC));
+
+        _clock = mock(Clock.class);
+        when(_clock.millis()).then(ignore -> _now.toEpochMilli());
+        when(_clock.instant()).then(ignore -> _now);
+    }
+
+    @AfterMethod
+    public void cleanUp() {
+        _uploadedAudits.clear();
+
+        if (_tempStagingDir.exists()) {
+            File[] files = _tempStagingDir.listFiles();
+            if (files != null) {
+                for (File file : files) {
+                    file.delete();
+                }
+            }
+            _tempStagingDir.delete();
+        }
+    }
+
+    private AthenaAuditWriter createWriter(String s3Path, String prefix, long maxFileSize, Duration maxBatchTime) {
+
+        _auditService = mock(ScheduledExecutorService.class);
+        _fileTransferService = mock(ExecutorService.class);
+
+        AthenaAuditWriter writer = new AthenaAuditWriter(_s3, BUCKET, s3Path, maxFileSize,
+                maxBatchTime, _tempStagingDir, prefix, Jackson.newObjectMapper(), _clock,
+                mock(GracefulShutdownRegistry.class), _auditService, _fileTransferService);
+
+        // On start two services should have been submitted: one to poll the audit queue and one to close log files and
+        // initiate transfers.  Capture them now.
+
+        ArgumentCaptor<Runnable> processQueuedAudits = ArgumentCaptor.forClass(Runnable.class);
+        verify(_auditService).scheduleWithFixedDelay(processQueuedAudits.capture(), eq(0L), eq(1L), eq(TimeUnit.SECONDS));
+        _processQueuedAudits = processQueuedAudits.getValue();
+
+        ArgumentCaptor<Runnable> doLogFileMaintenance = ArgumentCaptor.forClass(Runnable.class);
+        verify(_auditService).scheduleAtFixedRate(
+                doLogFileMaintenance.capture(),
+                longThat(new LessThan<>(maxBatchTime.toMillis()+1)),
+                eq(maxBatchTime.toMillis()), eq(TimeUnit.MILLISECONDS));
+        _doLogFileMaintenance = doLogFileMaintenance.getValue();
+
+        // Normally should avoid resetting mocks but it's already embedded in the writer.
+        reset(_auditService);
+
+        return writer;
+    }
+
+    @Test
+    public void testSimpleAudits() throws Exception {
```

Comment:
May be worth adding a few deeper unit tests, specifically to check

1. Audits which appear out of order still get written to the correct files.
2. Verify the graceful shutdown does indeed work as expected (writing stops, drain the queue, deliver all remaining audits).
3. More targeted test(s) that stopping writes works as expected, including when the shutdown happens mid-update-all.

Some of the above may be better suited to a different test class, but there should be tests for them somewhere.